File: factorial.py
import sys

sys.set_int_max_str_digits(0)

def factorial(n):
    result = 1
    for i in range(1, n + 1):
        result *= i
    return result

def print_usage():
    print('Usage: factorial.py N, where N is the number you want to get the factorial of.\n', file=sys.stderr)

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print_usage()
        sys.exit(1)
    try:
        n = int(sys.argv[1])
    except ValueError:
        print_usage()
        sys.exit(1)
    result = factorial(n)
    print(f'factorial({n}) = {result}')

File: factorial.c
#include <gmp.h>
#include <stdio.h>
#include <stdlib.h>

void factorial(mpz_t result, long n)
{
    mpz_init_set_si(result, 1);
    for (long i = 1; i <= n; ++i) {
        mpz_mul_si(result, result, i);
    }
}

void print_usage()
{
    fprintf(stderr, "Usage: factorial N, where N is the number you want to get the factorial of.\n");
}

int main(int argc, char** argv)
{
    if (argc != 2) {
        print_usage();
        return 1;
    }
    char* end = NULL;
    long n = strtol(argv[1], &end, 10);
    if (*end != '\0') {
        print_usage();
        return 1;
    }
    mpz_t result;
    factorial(result, n);
    gmp_printf("factorial(%d) = %Zd\n", n, result);
    mpz_clear(result);
}
